examples/simple-generator/src/app/index.ts

```diff
@@ -0,0 +1,142 @@
+import { getServiceInfo, getServiceInfoInBAS } from './service';
+
+export default class extends Generator {
+    private app!: Ui5App & { app: { flpAppId: string } };
+    private service!: OdataService;
+    private template: {
+        type: TemplateType | FreestyleTemplateType.Basic;
+        settings?: LROPSettings | OVPSettings | {};
+    };
+
+    initializing(): void {
+        this.log(
+            'Example of a simple Fiori elements for OData v4 generator that only creates listreport objectpage applications.'
+        );
+    }
+
+    async prompting(): Promise<void> {
+        const { name, template } = await this.prompt([
+            {
+                type: 'input',
+                name: 'name',
+                message: 'Application name',
+                validate: (answer) => !!answer
+            },
+            {
+                type: 'list',
+                name: 'template',
+                message: 'Template',
+                choices: [
+                    { value: TemplateType.ListReportObjectPage, name: 'List Report' },
+                    { value: TemplateType.OverviewPage, name: 'Overview Page' },
+                    { value: FreestyleTemplateType.Basic, name: 'Freestyle App' }
+                ]
+            }
+        ]);
+
+        let service: ServiceInfo;
+        if (isAppStudio()) {
+            service = await getServiceInfoInBAS(this);
+        } else {
+            service = await getServiceInfo(this);
+        }
+
+        const { entity } = await this.prompt({
+            type: 'input',
+            name: 'entity',
+            message: 'Main entity',
+            validate: (answer) => !!answer
+        });
+
+        this.service = {
+            version: OdataVersion.v2,
+            url: service.url,
+            path: service.path,
+            destination: {
+                name: service.destination
+            },
+            metadata: service.metadata
+        };
+        this.app = {
+            app: {
+                id: name,
+                flpAppId: 'app-preview'
+            },
+            appOptions: {
+                loadReuseLibs: true
+            },
+            package: {
+                name
+            }
+        };
+        this.template = {
+            type: template
+        };
+        switch (template) {
+            case TemplateType.OverviewPage:
+                this.template.settings = {
+                    filterEntityType: entity
+                } as OVPSettings;
+                break;
+            case TemplateType.ListReportObjectPage:
+                this.template.settings = {
+                    entityConfig: {
+                        mainEntityName: entity
+                    }
+                } as LROPSettings;
+                break;
+            default:
+                this.template.settings = {};
+                break;
+        }
+    }
+
+    configuring() {
+        this.sourceRoot(join(__dirname, '..', '..', 'templates', 'karma'));
```

Comment:
Add a comment explaining the reason for the path mangling?

Reply:
Done https://github.com/SAP/open-ux-tools/pull/466/commits/2d2c2a4316c8a4eca839b5f0ccefc4dc13f44de8